src/components/chips/js/chipsDirective.js

```diff
@@ -67,11 +67,16 @@
    *    displayed when there is at least on item in the list
    * @param {boolean=} readonly Disables list manipulation (deleting or adding list items), hiding
    *    the input and delete buttons
-   * @param {expression} md-on-append An expression that when called expects you to return an object
-   *    representation of the chip input string.
+   * @param {expression} md-on-append An expression of form `myFunction($chip)` that when called
```

Comment:
The `md-on-select` could indeed be used, but that forces the user to watch the list for changes to see whether the item was added or just selected by the user.

Additionally (and more importantly), there is currently no way to use the autocomplete functionality along with the "add chip" functionality, so if users attempt to do this, they wind up getting the chip from the autocomplete AND a blank/empty chip (#4412).